WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/domainregistration/DomainRegistrationSource.kt

```diff
@@ -74,25 +75,32 @@ class DomainRegistrationSource @Inject constructor(
         siteLocalId: Int,
         selectedSite: SiteModel
     ) {
-        continuation?.cancel()
-        continuation = null
-        coroutineScope.launch(bgDispatcher) {
-            try {
-                val event = suspendCancellableCoroutine<OnPlansFetched> { cancellableContinuation ->
-                    continuation = cancellableContinuation
-                    fetchPlans(selectedSite)
-                }
-                continuation = null
-                if (event.isError) {
-                    val message = "An error occurred while fetching plans : " + event.error.message
-                    appLogWrapper.e(DOMAIN_REGISTRATION, message)
+        if (continuations[siteLocalId] == null) {
```

Comment:
1. Praise (❤️): Great work here!
2. Question (❓): Are we sure that cancelling a continuation is not needed at all anymore?
3. Suggestion (💡): What if the `try/catch` gets extracted into its own `fetchPlans(...)` suspend function to make this `continuations` solution a little bit more visible? For example:
```
        if (continuations[siteLocalId] == null) {
            coroutineScope.launch(bgDispatcher) { fetchPlans(siteLocalId, selectedSite) }
        } else {
            appLogWrapper.d(DOMAIN_REGISTRATION, "A request is already running for $siteLocalId")
        }
```

Reply:
> Question (❓): Are we sure that cancelling a continuation is not needed at all anymore?

Not directly, but please elaborate if you feel there is a situation I could have missed. 

> Suggestion (💡): What if the try/catch gets extracted into its own fetchPlans(...) suspend function to make this continuations solution a little bit more visible?

Excellent idea in a839061

Reply:
👋 @zwarm !

> Not directly, but please elaborate if you feel there is a situation I could have missed.

Yeap, I am not sure as well, just making sure we are not missing something... 😅